include/SugarObjects/templates/person/Person.php

```diff
@@ -333,4 +337,45 @@ public function create_export_query($order_by, $where, $relate_link_join = '')
         return $query;
     }
 
+    /**
+     * Set Lawful Basis
+     * @param string $basis
+     * @param string $source
+     * @return int
+     * @throw InvalidArgumentException
+     */
+    public function setLawfulBasis($basis, $source)
+    {
+        global $app_list_strings,$timedate;
+        /**
+         * This function will update the lawful basis, source and date of the change.
+         * Will take the parameters of email id and possible the module?
+         */
+
+        if (!is_string($basis)) {
+            throw new InvalidArgumentException('basis must be a string');
+        }
+
+        if (!array_key_exists($basis, $app_list_strings['lawful_basiss'])) {
+            throw new InvalidArgumentException('invalid lawful basis');
+        }
+
+        if (!is_string($source)) {
+            throw new InvalidArgumentException('source for lawful basis must be a string');
+        }
+
+        if (!array_key_exists($source, $app_list_strings['lawful_basis_sources'])) {
+            throw new InvalidArgumentException('invalid lawful basis source');
+        }
+
+        //Set lawful basis, lawful basis source and date reviewed
+        $this->lawful_basis = '^'.$basis.'^';
+        $this->lawful_basis_source = $source;
+        $date = TimeDate::getInstance()->nowDb();
+        $date_test = $timedate->to_display_date($date,false);
+        $this->date_reviewed = $date_test;
+        $this->save();
+
+        return 0;
```

Comment:
It seems the return value is not used, but I'd argue that if something is returned after a successful operation it be 1 or true and not 0.